tests/gpflow/optimizers/test_mcmc.py

```diff
@@ -199,7 +196,7 @@ def run_chain_fn():
         assert hmc_helper._parameters[i].numpy() == parameter_samples[i][-1]
 
 
-@pytest.mark.xfail(raises=ValueError)
```

Comment:
`xfail` is meant for tests that you aren't expecting to pass - e.g. for tests of not-yet-implemented features. To test that code raises an exception as expected (e.g. error handling), use `with pytest.raises()` (see below) - this also allows regexp-matching to check it's got the right error message!